Rotation

COBRA445 opened the pull request.

COBRA445 committed:
added rotation + refactoring

COBRA445 committed:
First-person view

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+    // Start is called before the first frame update
+    /// <summary>
+    //TODO �������� ���������� ������ � ����. ������ - ��� ������ �����/����
+    //TODO �������� � UI ����� ��������� ���� � ����� + �����? ���?
```
Посмотри на кодировку. Должна быть  UTF-8. Это стандарт.
Погугли, как "настроить кодировку в VisualStudio"

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+    public GameObject Cube;
+    public GameObject CubeGenerator;
+
+    public cubeScript cubeScript;
```
Наименование не по феншую. 2 момента:
1. Наименование скрипта `cubeScript`: наименование скриптов с большой буквы. 
2. Наименование переменной `cubeScript`: так же с большой буквы, так как она публичная

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+    [SerializeField]
+    private int score = 0;
+    [SerializeField]
+    private int randomX_Highest = 25;
```
Наименование переменных некорректно. См. доку

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+
+
+    private void Start() {
+        spawnCubeGeneator();
```
Наименование функции - некорректное

COBRA445 replied:
Я поправил это...видимо не залил изменения(хотя вроде заливал...)

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+        float randomY = transform.position.y;
+        float randomZ = transform.position.z;
+
+        Instantiate(CubeGenerator, new Vector3(UnityEngine.Random.Range(randomX_Highest, randomX_Lowest), 0, UnityEngine.Random.Range(randomZ_Highest, randomZ_Lowest)), transform.rotation);
```
Если есть риск перепутать с другим пространством имен (например, `UnityEngine.Random` и `System.Random`),  то в начали скрипта можно указать конкретный вариант использования (например, `using Random = UnityEngine.Random;`

Gr871 replied:
Это делается для того, чтобы в скриптах не тянуть длинные названия

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+
+    public void GameOver() {
+        start = false;
+        UIScript.GameOver(score);
```
`UIScript` не по феншую. В аббревиатурах с Заглавной буквы идет только первая (так что, `UiScript`)

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,94 @@
+
+    }
+
+    public void restartGame() {
```
не по феншую

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+    private Color colorText = new Color(0.1960784f, 0.1960784f, 0.1960784f); //505050
+                                                                             // Start is called before the first frame update
+
+    private int score = 0;
+    private float timesTimerAll = 0;  //������� �������
+    private float timesTimer = 0;  //���������� ��� ��������
```
Наименование приватных переменных не по феншую (должно начинаться с `_`, например,  `_score`)

Gr871 replied:
+ Смотри, что с комментариями. Опять кодировка не корректная

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+    }
+
+
+    public void Update() {
+
+    }
```
Если функция бесполезная  - не нужно ее объявлять. Особенно это касается функций `Monobehaviour`'a

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+    public GameObject GameOverButton;
+    public GameObject StartButton;
+
+    public Text textScore;
+    public Text textTimer;
+    public Text textResult;
+    public Text textBestRresults;
```
Не по феншую

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+        for (int i = 0; i < scoreTopResult.number.Length; i++) {
+            if (scoreTopResult.time[i] < 10) {
+                result += scoreTopResult.number[i] + ". " + scoreTopResult.scoreResult[i] + "   (" + (int)(scoreTopResult.time[i] / 60f) + ":0" + (int)(scoreTopResult.time[i] % 60f) + ")\n";
+            }
+            else {
```
в одну строку

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+        textBestRresults.text = result;
+    }
+
+    public void readTopResultFromFiles() {
+        scoreTopResult = JsonUtility.FromJson<ScoreTopResult>(File.ReadAllText(Application.streamingAssetsPath + "/JSON_Result.json"));
+    }
+    public void writeTopResultFromFiles() {
+        searchBestResult();
+        File.WriteAllText(Application.streamingAssetsPath + "/JSON_Result.json", JsonUtility.ToJson(scoreTopResult));
+        BestRresults();
+    }
+    public void searchBestResult() {
```
Разделяй минимум одной строкой объявления фнукций. А желаетльно, еще комментарии добавлять к каждой

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,95 @@
+
+    }
+
+    public void cube_Move(bool can) {
```
Наименование переменной

COBRA445 replied:
Тоже пофиксил, но не залил

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,95 @@
+        if (can) {
+            if (Input.GetKey(KeyCode.W)) {
+                //  transform.position = new Vector3(transform.position.x + speed * Time.deltaTime, transform.position.y, transform.position.z + speed * Time.deltaTime)
+                transform.position = new Vector3(transform.position.x + transform.forward.x * speed * Time.deltaTime,
+                                                 transform.position.y,
+                                                 transform.position.z + transform.forward.z * speed * Time.deltaTime);
```
С выравниванием беда

COBRA445 replied:
Почему? Я же здесь специально так разделил, чтою переменные (передаваемые в конструктор) были друг под другом

Gr871 replied:
не, сейчас все ровно
этот комментарий, видимо, еще от старого варианта остался

Когда исправил - жми Resolve, чтобы таких эксцессов не было

Gr871 commented on Assets/Scripts/cubeSpawnerScrip.cs:
```diff
@@ -1,24 +1,18 @@
 
-public class cubeSpawnerScrip : MonoBehaviour
-{
+public class cubeSpawnerScrip : MonoBehaviour {
```
Наименование класса не по феншую

Gr871 commented on Assets/Scripts/cubeSpawnerScrip.cs:
```diff
@@ -1,24 +1,18 @@
-    void Update()
-    {
-        
+    void Update() {
```
пустой скрипт ни к чему

COBRA445 committed:
added a hot-cold indicator to the game depending

 on the distance and angle of rotation (Interpolation: Linear interpolation, Inverse Lerp, Remap, Slerp)

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,146 @@
+    [SerializeField]
+    private bool clone = false;   
+
+    private float kDistance = 0;    
+    private float kRotation = 0;
```
приватные несериализуемые перменные начинаются с "_": `_kDistance`

Gr871 replied:
второй момент что `k` вносит некий деструктив в именование. Я так понимаю, что речь идет о "коэффициенте", которые в переводе на английский - `ratio` или `aspect`

Gr871 replied:
должно получиться, например, `_distanseAspect` или `_distanceRatio`.

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,146 @@
+    private float kDistance = 0;    
+    private float kRotation = 0;
+    [SerializeField]
+    private float angleBeetwenCubeAndCubeClone = 0;
```
А зачем тебе эта переменная? я, во-первых, не вижу ее использование, а во-вторых, сериализация нужна тогда, когда ты хочешь изменять параметр из эдитора руками (ну или в крайнем случае вывести для отладки в него же). Думаю, что здесь тебе "УЖЕ" это не требуется

Gr871 commented on Assets/Scripts/LogicScript.cs:
```diff
@@ -0,0 +1,146 @@
+
+
+    public GameObject Cube;
+    public GameObject CubeGenerator;
+    private GameObject CubeGeneratorClone;
```
Давай разбираться с наименованиями:
1. Что такого `CubeGenerator`? из названия следует, что это объект, который самостоятельно решает задачу генерации каких-то кубов. А я подозреваю, что это лишь "основа" для генерации **цели** (**префаб**, с которого скрипт в одной из функци ниже создает копию). Так тогда давай и переименуем эту переменную в `TargetSourcePrefab`, например, и будет понятно, что это "префаб-источник" для создания цели.
2. а зачем тебе эта переменная публичная? может достаточно сделать ее приватной и `[SerializedField]`?

Gr871 replied:
Отсюда `CubeGeneratorClone` (к слову, напоминаю, что приватные переменные именуются не так) название не корректное. Суффикс `Clone`, в целом, как-то да отражает суть. Но куда логичнее вместо `Clone` использовать `Instance` и тогда однозначно понятно, что `targetInstance` - это "цель", конкретный "экземпляр" объекта `TargetSourcePrefab`

Gr871 commented on Assets/Scripts/MathLib/Interpolation.cs:
```diff
@@ -0,0 +1,72 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Interpolation {
```
красота. Вроде как по всем канонам)

внизу тока подрежь кол-во пустых строк

Gr871 commented on Assets/Scripts/MathLib/Trigonometry.cs:
```diff
@@ -0,0 +1,3 @@
+﻿public class Trigonometry {
```
ну думаю, что нужен этот скрипт

Gr871 commented on Assets/Scripts/MathLib/Vector3D.cs:
```diff
@@ -0,0 +1,304 @@
+            float scalingVecotor = vectorA.X * vectorB.X + vectorA.Y * vectorB.Y + vectorA.Z * vectorB.Z;
+            return scalingVecotor;
+        }
+        /*
+        public static float ScalingVector3Dcos(Vector3D vectorA, Vector3D vectorB) { //скалярное умножение через косинус - не нужно   //|a| × |b| × cos(θ)    
+            float scalingVecotor = Length(vectorA) * Length(vectorB) * CosVector3D(vectorA, vectorB);
+            return scalingVecotor;
+        }
+        */
```
Выпиливай, что не используется
А так - тоже конфетка

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+        }
+        else {
+            textTimer.color = colorText;
+            //colorText = new Color(red, green, blue);
+            // Debug.Log("text color " + red + "  " + green + "  " + blue); // ������ ���� ����������� ���������� � ������ ���� - ������ 
```
не нужные строки - удаляй

Gr871 commented on Assets/Scripts/UIScript.cs:
```diff
@@ -0,0 +1,165 @@
+                scoreTopResult.time[i] = time_temporary2;
+                score_result = score_temporary;
+                time_temporary2 = time_temporary;
+                //  Debug.log("score: " + score);
+                //  Debug.log("THE_BEST_res: " + scoreTopResult.scoreResult[i]);
+                //  Debug.log("TIME: " + time_temporary2);
```
не нужные строки

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,108 @@
+using UnityEngine;
+
+public class cubeScript : MonoBehaviour {
```
Наименование скрипта

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,108 @@
+
+        GameOverBorders();
+        if (can) {
+            �ubeMove(can);
+        }
+        logic.�hanging�olor();
```
ну тут какие-то фантастические имена

Gr871 replied:
а, знаю: это `C` русская

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,108 @@
+            if (Input.GetKey(KeyCode.A)) {
+                //transform.position = new Vector3(transform.position.x - speed * Time.deltaTime, transform.position.y, transform.position.z + speed * Time.deltaTime);
+                RotationCubeLeftX(gameObject);
+                //     transform.position = RotationX(gameObject);
+
+
+
```
пустые и ненужные строки

Gr871 commented on Assets/Scripts/cubeScript.cs:
```diff
@@ -0,0 +1,108 @@
+
+
+    }
+    //public void startPosition() {
+    //    _rigidbody = GetComponent<Rigidbody>();     
+    //    _rigidbody.mass = 0f;
+    //    transform.position = new Vector3(0, 0, 0);
+    //    transform.localScale = new Vector3(1, 1, 1);
+    //    _rigidbody.mass = 1f;
+
+    //}
```
не нужные строки

Gr871 commented on Assets/Scripts/indicator.cs:
```diff
@@ -0,0 +1,77 @@
+        //renderer�ircle.material.color = colorObject;
+    }
+
+    public void �hanging�olorDistance(float distance, float rotation) {
```
Как ты умудряешься такие именования фигачить?)

COBRA445 replied:
Что-то я сам в шоке....

Gr871 commented on Assets/Scripts/indicator.cs:
```diff
@@ -0,0 +1,77 @@
+
+
+
+        //var colors = button.GetComponent<Button>().colors;
+        //colors.normalColor = Color.Lerp(colorHot, colorCold, distance);
+        //colors.highlightedColor = Color.Lerp(colorHot, colorCold, distance);
+        //colors.pressedColor = Color.Lerp(colorHot, colorCold, distance);
+        //colors.selectedColor = Color.Lerp(colorHot, colorCold, distance);
+        //colors.disabledColor = Color.Lerp(colorHot, colorCold, distance);
+        //button.GetComponent<Button>().colors = colors;
+
+
+        //button.GetComponent<Image>().color = color;
```
пустые строки
